code/game/machinery/firealarm.dm

```diff
@@ -237,11 +237,16 @@ FIRE ALARM
 	else
 		set_light(l_power = LIGHTING_MINIMUM_POWER)
 
+/obj/machinery/firealarm/proc/update_fire_sound(fire)
+	if(stat & NOPOWER)
```

Comment:
Do you want to start the sound again if power returns?

Reply:
I wasn't sure if I should but sure, new commit